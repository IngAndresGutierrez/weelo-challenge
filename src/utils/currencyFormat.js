/**
 * return value currency format
 * @param   {number} amount Value for format
 */

export const addCurrencyFormatUsd = (amount) => {
  const optionCurrencyUSD = { style: 'currency', currency: 'USD' }
  const numberPriceUsdFormat = new Intl.NumberFormat('en-US', optionCurrencyUSD)
  const priceUsdFormat = numberPriceUsdFormat.format(amount)

  return priceUsdFormat
}

/**
 * return value currency format without dollar symbol
 * @param   {number} amount Value for format
 */
export const addCurrencyFormatUsdWithoutDollarSymbol = (amount) => {
  const optionCurrencyUSD = { currency: 'USD' }
  const numberPriceUsdFormat = new Intl.NumberFormat('en-US', optionCurrencyUSD)
  const priceUsdFormat = numberPriceUsdFormat.format(amount)

  return priceUsdFormat
}
