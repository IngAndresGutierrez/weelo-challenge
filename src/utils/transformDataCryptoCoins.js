/**
 * return all information into rows table
 * @param   {cryptoCoin[]} coinsList Information about all coins
 */

export const transformData = (coinsList) => {
  const newRowsCoins = []
  coinsList?.forEach((coin) => {
    let newCoin = createData(
      coin.id,
      coin.symbol,
      coin.name,
      coin.rank,
      coin.price_usd,
      coin.market_cap_usd,
      coin.percent_change_24h,
      coin.percent_change_1h,
      coin.percent_change_7d,
      coin.price_btc,
      coin.volume24,
      coin.volume24a,
      coin.csupply,
      coin.tsupply,
      coin.msupply
    )

    newRowsCoins.push(newCoin)
  })

  return newRowsCoins
}

/**
 * return all information about one coin into row table
 * @param   {infoCryptoCoin} infoCryptoCoin Information about one coin
 */

const createData = (
  id,
  symbol,
  name,
  rank,
  price_usd,
  market_cap_usd,
  percent_change_24h,
  percent_change_1h,
  percent_change_7d,
  price_btc,
  volume24,
  volume24a,
  csupply,
  tsupply,
  msupply
) => {
  return {
    id,
    symbol,
    name,
    rank,
    price_usd,
    market_cap_usd,
    percent_change_24h,
    percent_change_1h,
    percent_change_7d,
    price_btc,
    volume24,
    volume24a,
    csupply,
    tsupply,
    msupply,
  }
}
