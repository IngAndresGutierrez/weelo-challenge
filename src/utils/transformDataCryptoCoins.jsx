export const transformData = (coinsList) => {
  const newRowsCoins = []
  coinsList?.forEach((coin) => {
    let newCoin = createData(
      coin.symbol,
      coin.name,
      coin.rank,
      coin.price_usd,
      coin.market_cap_usd
    )

    newRowsCoins.push(newCoin)
  })

  return newRowsCoins
}

const createData = (symbol, name, rank, price_usd, market_cap_usd) => {
  return { symbol, name, rank, price_usd, market_cap_usd }
}
