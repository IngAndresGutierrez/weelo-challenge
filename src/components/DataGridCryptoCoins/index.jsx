/* eslint-disable react-hooks/exhaustive-deps */
/* import external modules */
import { useEffect, useState } from 'react'
import { DataGrid } from '@mui/x-data-grid'

/* import internal modules */
import { transformData } from '../../utils/transformDataCryptoCoins'

const columns = [
  { field: 'symbol', headerName: 'Symbol', width: 140 },
  {
    field: 'name',
    headerName: 'Name',
    width: 190,
  },
  {
    field: 'rank',
    headerName: 'Rank',
    width: 190,
  },
  {
    field: 'price_usd',
    headerName: 'Price USD',
    width: 210,
  },
  {
    field: 'market_cap_usd',
    headerName: 'Market Cap USD',
    width: 210,
  },
  {
    field: 'options',
    headerName: 'Options',
    width: 200,
  },
]

const DataGridCryptoCoins = ({ coinsList }) => {
  const [rows, setRows] = useState([])

  useEffect(() => {
    transFormDataFunction()
  }, [coinsList])

  const transFormDataFunction = () => {
    if (coinsList) {
      const transformRows = transformData(coinsList)
      setRows(transformRows)
    }
  }

  return (
    <div style={{ height: 400, width: '100%' }}>
      <DataGrid
        rows={rows}
        pageSize={5}
        columns={columns}
        rowsPerPageOptions={[5]}
        disableSelectionOnClick
      />
    </div>
  )
}

export default DataGridCryptoCoins
