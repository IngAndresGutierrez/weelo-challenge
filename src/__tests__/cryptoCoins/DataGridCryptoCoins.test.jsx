/* import external modules */
import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

/* import internal modules */
import '../../../setupTests'
import { cryptoCoinsList } from '../../utils/cryptoCoinsList'
import DataGridCryptoCoins from '../../components/DataGridCryptoCoins'

describe('Testing suite DataGridCryptoCoins component', () => {
  const DataGridCryptoCoinsShallow = shallow(<DataGridCryptoCoins />)
  const DataGridCryptoCoinsMount = mount(
    <DataGridCryptoCoins coinsList={cryptoCoinsList} />
  )

  test('Render DataGridCryptoCoins', () => {
    expect(DataGridCryptoCoinsShallow).toHaveLength(1)
  })

  test('Renders correctly - Snapshot', () => {
    const treeDataGridCryptoCoinsMount = renderer
      .create(DataGridCryptoCoinsMount)
      .toJSON()
    expect(treeDataGridCryptoCoinsMount).toMatchSnapshot()
  })

  test('Render texts in Toolbar', () => {
    expect(DataGridCryptoCoinsMount.find('span').first().text()).toEqual(
      'Columns'
    )
  })

  test('Render data from crypto coins list', () => {
    // console.log(DataGridCryptoCoinsMount.debug()) /* Debug */
    expect(DataGridCryptoCoinsMount.find('div').at(16).text()).toEqual('Name')
    expect(DataGridCryptoCoinsMount.find('div').at(36).text()).toEqual(
      '1-5 of 100'
    )
  })
})
