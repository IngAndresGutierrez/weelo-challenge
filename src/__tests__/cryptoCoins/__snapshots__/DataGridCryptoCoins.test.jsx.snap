// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing suite DataGridCryptoCoins component Renders correctly - Snapshot 1`] = `
<div
  style={
    Object {
      "height": 420,
      "width": "100%",
    }
  }
/>
`;
